### PR TITLE
Implement domain of Review micro-service

### LINKED ISSUE #14: Implement domain classes and unit tests
**Description**
Core business logic of the Review micro-service

**Tasks**
- [ ] Add domain classes
- [ ] Add unit tests 
- [ ] Set up Hexagonal skeleton

### PR DESCRIPTION
**Pull request by xandervedder**
## Changes
Add a few domain classes that will eventually make it possible to save reviews.

## Checklist
- [x] Add `Review` class with unit tests
- [x] Format all modified files
- [x] Check for missing newlines

## When merging, this pull request 
closes #14

### CODE REVIEW

File: review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+/**
+ * Class that contains data and logic to make a {@link Review} on a {@link ReviewType#PRODUCT} or
+ * {@link ReviewType#DELIVERY}.
+ */
+public class Review {
+    private static final int MIN_TITLE_LENGTH = 3; // Allows for titles like 'Bad' and 'Good'
+    private static final int MAX_TITLE_LENGTH = 32; // Disallow large sentences (those are meant for the description)
+    private static final int MIN_DESCRIPTION_LENGTH = 32;
+    private static final int MAX_DESCRIPTION_LENGTH = 256; // Is this not enough? Should it be more?
```
Comment (Rubenvdbrink) [inline review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java:14]: I would say 512 characters

**Comment by JustMilan** [comment]
Why use a .gitkeep to keep the application folder if it's not being used?

**Comment by xandervedder** [comment]
> Why use a .gitkeep to keep the application folder if it's not being used?

So I don't have to create the folder again 🧠

File: review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    public UUID getId() {
+        return id;
+    }
+
+    /**
+     * Sets the customers' id of the {@link Review}.
+     *
+     * @param customerId The id of the customer.
```
Comment (JustMilan) [inline review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java:55]: nit: Here, you are using a capital letter after the param name while you're using non capital letters in other @param

Comment (JustMilan) [reply review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java:55]: There are multiple occasions where this happens.

**Review by JustMilan — CHANGES_REQUESTED** [review]

File: review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java (hunk 2 above)
Comment (JustMilan) [reply review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java:55]: If you don't see this as any sort of problem, just resolve this.

**Review by Rubenvdbrink — DISMISSED** [review]
LGTM

**Review by JustMilan — CHANGES_REQUESTED** [review]
Please change PR title as you've said here
https://discord.com/channels/679327408343613460/684494959054684236/915365806479638528

File: review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java (hunk 2 above)
Comment (xandervedder) [reply review/src/main/java/nl/softwarestrijders/waiter/review/core/domain/Review.java:55]: Should be consistent now

**Comment by xandervedder** [comment]
Not done yet, I have differentiate between a product and delivery review i.e. i need the product id or delivery id... will continue working on this a bit later

**Review by Rubenvdbrink — APPROVED** [review]
👀✅2me

**Review by JustMilan — APPROVED** [review]

**Pull request merged**